src/components/Island.tsx

```diff
@@ -2,16 +2,27 @@ import "./Island.css";
 
 import React from "react";
 
-type IslandProps = { children: React.ReactNode; padding?: number };
+type IslandProps = {
+  children: React.ReactNode;
+  padding?: number;
+  maxHeight?: number;
+};
 
 export const Island = React.forwardRef<HTMLDivElement, IslandProps>(
-  ({ children, padding }, ref) => (
-    <div
-      className="Island"
-      style={{ "--padding": padding } as React.CSSProperties}
-      ref={ref}
-    >
-      {children}
-    </div>
-  ),
+  ({ children, padding, maxHeight }, ref) => {
+    const style = {
+      "--padding": padding,
+    } as React.CSSProperties;
+
+    if (maxHeight && maxHeight > 0) {
+      style.maxHeight = `${maxHeight}px`;
+      style.overflowY = "scroll";
```

Comment:
This should be `auto` for systems where scroll bars are always visible.

```suggestion
      style.overflowY = "auto";
```